### ISSUE TITLE
prognostic stand structure failure

### ISSUE THREAD

**Issue by pollybuotte** [NGEET/fates#548]
I have cases on Lawrencium (not sure if the machine is important; I can't test it on Cheyenne this week) that will run with initialized, static stand structure, but fail with this error in the cesm.log file when setting use_fates_ed_st3 = .false., regardless of the use_fates_inventory_init setting:

forrtl: error (65): floating invalid
Image PC Routine Line Source
cesm.exe 000000000390E21D Unknown Unknown Unknown
cesm.exe 000000000390C0B7 Unknown Unknown Unknown
cesm.exe 00000000038AE9F4 Unknown Unknown Unknown
cesm.exe 00000000038AE806 Unknown Unknown Unknown
cesm.exe 0000000003831976 Unknown Unknown Unknown
cesm.exe 000000000383C747 Unknown Unknown Unknown
Unknown 00002ACA5B2315D0 Unknown Unknown Unknown
cesm.exe 00000000038AD43B Unknown Unknown Unknown
cesm.exe 00000000013AEDD6 fateshistoryinter 1651 FatesHistoryInterfaceMod.F90
cesm.exe 0000000000916047 clmfatesinterface 1295 clmfates_interfaceMod.F90
cesm.exe 000000000088BBA6 clm_initializemod 667 clm_initializeMod.F90
cesm.exe 0000000000821D3A lnd_comp_mct_mp_l 232 lnd_comp_mct.F90
cesm.exe 0000000000459ECC component_mod_mp_ 246 component_mod.F90
cesm.exe 0000000000422149 cime_comp_mod_mp_ 1243 cime_comp_mod.F90
cesm.exe 00000000004516AC MAIN__ 122 cime_driver.F90
cesm.exe 0000000000411E5E Unknown Unknown Unknown
libc.so.6 00002ACA5B4603D5 Unknown Unknown Unknown
cesm.exe 0000000000411D69 Unknown Unknown Unknown
mpirun noticed that process rank 0 with PID 13378 on node n0064.lr3 exited on signal 6 (Aborted).
__________________________________________________________________________________
The model date is being passed from the host land model to FATES in line 1651 of FatesHistoryInterfaceMod.

The lnd.log file indicates no errors:

There will be a total of 1 history tapes
History tape 1 write frequency = 17520
All fields on history tape 1 are grid averaged
Number of time samples on history tape 1 is 31
Output precision on history tape 1 = 2
hist_htapes_build hist tape = 1 written with dov2xy= T
hist_htapes_build Successfully initialized clm2 history files
_______________________________________________________________

I am running with this tag:
commit 6232bb3
Merge: 0cf63a9 8efc24f
Author: Ryan Knox rgknox@lbl.gov
Date: Fri Jun 14 11:00:28 2019 -0700

**Comment by rgknox** [comment]
Working with @pollybuotte to test on lawrencium.  Currently waiting on a help ticket from their HPC help to resolve a problem with their module system.

**Comment by glemieux** [comment]
@pollybuotte I saw the same failure modes on the regression test off of the fates master branch that I ran a few days ago to try and get used to running on lawrencium.  

@rgknox was the error message you saw wrt the modules something like:
>Failed to load module "canberra-gtk-module"
I got the above trying to fire up emacs.

**Comment by glemieux** [comment]
Also @rgknox, I'm seeing the same FAIL as you noted in https://github.com/ESCOMP/ctsm/issues/322.  It looks like this didn't get merged into fates_next_api?

**Comment by rgknox** [comment]
The module problem I had on Lawrencium seems to have been an aberration.  The  canberra-gtk-module things does not look too familiar.

Regarding the Luna fix.  Yeah, lets get that in there. Although, hopefuly we will resolve how we want to sync fates_next_api with CTSM branches, and this luna problem will be resolved in that process.

**Comment by glemieux** [comment]
> ```
> There are not enough slots available in the system to satisfy the 16 slots
> that were requested by the application:
> /global/scratch/glemieux/ERP_D_P32x2_Ld3.f19_g16.I2000Clm50FatesCruGs.lawrenciumlr3_intel.clm-FatesColdDef.G.20190624_163133_g6bg8i/bld/cesm.exe
> Either request fewer slots for your application, or make more slots available
> for use.
> ```

@rgknox the command that is failing is `mpirun -np 32  -npernode 16`, but if I understand correctly, shouldn't this be fine on lawrencium since there are 16 cores per node?  Or does 'slot' here not mean core in the mpi context?

**Comment by pollybuotte** [comment]
@glemieux @rgknox This looks like a Lawrencium issue.  I jut ran a successful case with prognostic stand structure, same tag, on Cheyenne.

**Comment by glemieux** [comment]
@pollybuotte can you share the particular case and/or script on lawrencium that failed?

**Comment by pollybuotte** [comment]
@glemieux here's the script, saved as .txt to be able to upload here.
[create_fates_single_pt_lr3.txt](https://github.com/NGEET/fates/files/3356429/create_fates_single_pt_lr3.txt)

**Comment by rgknox** [comment]
@glemieux , I have seen this error in the ERP_D_P32x2 test as well.  Although, I never got around to solving what the problem is.  It seems like it should be requesting two nodes with 16 threads on the 16 available cores per node.. but it doesn't seem to be doing that...?

**Comment by glemieux** [comment]
What's also interesting is that the `mpirun -np 32 -npernode 16` command is being called for other test runs in the same suite, both for debug and non-debug runs, some of which pass and others that don't.  I might dig around in the cime issues or the cesm forum to see if anyone else has seen this problem.  Out of curiousity, I also sent in an issue to hpcshelp@lbl.gov

**Comment by glemieux** [comment]
Update on the 'not enough slots' issue that we were seeing for the ERP_D_P32x2 regression test.  IT had me remove the `<arg name="tasks_per_node"> -npernode $MAX_MPITASKS_PER_NODE </arg>` from the config_machine.xml file for the lawrencium-lr3 section.  Their comments was:
> let slurm handle that since you have #SBATCH --ntasks-per-node=16 in your submission script.   

More details can be found in the [askUS@BerkeleyLab](https://lbl.service-now.com/lbl/home.do) site under the ticket ASKUSI00050552.

That regression test in particular is failing in a way similar to what @pollybuotte originally noted.  So that detour is kind of taking care of?  Now back to your regularly scheduled issue....

**Comment by glemieux** [comment]
@pollybuotte if you run your script with `./xmlchange DEBUG=FALSE` do you get the same error?

**Comment by pollybuotte** [comment]
@glemieux that fixed the error!  Dare I ask why?  Thanks very much.

**Comment by glemieux** [comment]
It looks like this is a problem in a similar vein to #549: I think the run is crashing because `hlm_model_day` is not being initialized before being called here:
https://github.com/NGEET/fates/blob/6cd0bb5cd93b5f338f410a1211d636989fe49154/main/FatesHistoryInterfaceMod.F90#L1641-L1652 
With the debugger off, this sets the value to 0, but with the debugger on it sets the value to NaN.

**Comment by glemieux** [comment]
@rgknox if this is really the case, any thoughts on where `hlm_model_day` should be initialized?  The subroutine encapsulating the above is `update_history_dyn` which I've determined is being called by `init_coldstart` in clmfates_interfaceMod.F90 (by adding some diagnostic output to the ctsm code):

https://github.com/glemieux/ctsm/blob/f0e3b950ffe91a2c60e634bc096746399e9c8eec/src/utils/clmfates_interfaceMod.F90#L1301-L1307

Would any of the fates functions called from `init_coldstart` be a good candidate (`zero_site`, `init_site_vars`, etc)?

**Comment by rgknox** [comment]
@glemieux and I just had a gathering of the minds, solutions forthcoming

**Comment by glemieux** [comment]
Fixed per CTSM#820
